### DISCUSSION TITLE
Improve Error Message for Missing JSON Keys when Loading into C++ Variable

### DISCUSSION THREAD

**Discussion by pwintz** [nlohmann/json#4443] [Ideas]
I think it would be helpful to improve the error messages when trying to read a key is not present in a `nlohmann::json` object. In particular, I have code like this:
```
  std::ifstream config_json_file(config_file_path);
  json json_data = json::parse(config_json_file);
  config_json_file.close();
  double tau = json_data["tau"];
```
If "tau" is not present in the JSON data, however, the following error is printed:
```
terminate called after throwing an instance of 'nlohmann::json_abi_v3_11_3::detail::type_error'
   what():  [json.exception.type_error.302] type must be number, but is null
```
This error doesn't give very helpful info for figuring out the problem. It would be helpful to print something like 
```
The key "tau" is not found. Expected it to be a number.
```

**Comment by nlohmann** [comment]
You may want to check out https://json.nlohmann.me/api/macros/json_diagnostics/
